src/main/java/gift/repository/ProductRepository.java

```diff
@@ -9,34 +14,62 @@
 @Repository
 public class ProductRepository {
 
-    private final Map<Long, Product> products = new HashMap<>();
+    private final JdbcTemplate jdbcTemplate;
+
+    public ProductRepository(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
     private final AtomicLong sequence = new AtomicLong(0);
 
+    private SimpleJdbcInsert insert;
+
+    @PostConstruct
+    public void init() {
+        this.insert = new SimpleJdbcInsert(jdbcTemplate)
+                .withTableName("product")
+                .usingGeneratedKeyColumns("id");
+    }
```

Comment:
별도의 `@PostConstruct`를 활용해서 SimpleJdbcInsert를 초기화해줄 필요 없이, 생성자를 통해서 초기화를 해주시면
SImpleJdbcTemplate 필드도 final로 선언할 수 있어 더 안전하고, `@PostConstruct`와 같은 스프링 의존도를 낮출 수 있을 것 같아요. 😃 


```suggestion
    private final JdbcTemplate jdbcTemplate;
    private final SimpleJdbcInsert insert;

    public ProductRepository(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
        this.insert = new SimpleJdbcInsert(jdbcTemplate)
            .withTableName("product")
            .usingGeneratedKeyColumns("id");
    }
```